provider/oci/common/paginatedclients.go

```diff
@@ -0,0 +1,349 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/oracle/oci-go-sdk/v47/common"
+	ociCore "github.com/oracle/oci-go-sdk/v47/core"
+)
+
+// NewComputeClient returns a client which implements the
+// OCIComputeClient interface.
+func NewComputeClient(provider common.ConfigurationProvider) (*computeClient, error) {
+	c, err := ociCore.NewComputeClientWithConfigurationProvider(provider)
+	return &computeClient{c}, err
+}
+
+type computeClient struct {
+	OCIComputeClient
```

Comment:
My inclination would be to make this a member so none of the public methods are hoisted, and wrap each method to unwind the pagination. So `PaginatedListImages` becomes `ListImages` again. Also `PaginatedListImages` suggests the results *are* paginated, rather than complete.

Reply:
Easy enough to do.